cardano-cli/src/Cardano/CLI/Mary/Parser.hs

```diff
@@ -0,0 +1,205 @@
+{-# LANGUAGE FlexibleContexts #-}
+
+module Cardano.CLI.Mary.Parser
+  ( Token (..)
+  , Tokens
+  , TParser
+  , addition
+  , calculateValue
+  , lexToken
+  , lexTokens
+  , preValueAddition
+  , preValueLovelace
+  , preValueMultiAsset
+  , preValueParser
+  , preValtoValue
+  , stringToValue
+  , textToPolicyId
+  , valueTokenFullySpecified
+  , valueTokenPidAndAssetId
+  , valueTokenPidOnly
+  ) where
+
+import           Cardano.Prelude hiding (try)
+import           Prelude (String, read)
+
+
+import           Control.Monad (fail)
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as Text
+import qualified Data.Text.Encoding as Text
+import           Text.Parsec (ParseError, Parsec, SourcePos, anyChar, getPosition, manyTill, parse,
+                     token, try, (<?>))
+import           Text.Parsec.Char (alphaNum, digit, hexDigit, letter, space, spaces, string)
+import           Text.Parsec.String (Parser)
+import           Text.ParserCombinators.Parsec.Combinator (eof, many1, notFollowedBy, sepBy1)
+
+import           Cardano.Crypto.Hash (hashFromStringAsHex)
+import           Cardano.Ledger.Mary (MaryEra)
+import           Cardano.Ledger.Mary.Value (AssetID (..), PolicyID (..), Value (..))
+import           Ouroboros.Consensus.Shelley.Eras (StandardCrypto)
+import           Shelley.Spec.Ledger.Scripts (ScriptHash (..))
+
+{- HLINT ignore "Reduce duplication" -}
```

Comment:
Is there good reason to ignore this? If so, perhaps leave a brief comment to explain why.

However, if it's a good `hlint` suggestion, I think we should address it.